File: game_client/Assets/Scripts/ClientProtocol/Common/LevyEnum/ELevy_Type.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.LevyEnum
{

/// <summary>
/// 征收类型类型
/// </summary>
public enum ELevy_Type {
	NONE, //0 ==== 
	[InspectorName("SILVER - idx[1] - 银币 -> 金币")]
	SILVER, //1 ==== 银币 -> 金币
	[InspectorName("FOOD - idx[2] - 粮食 -> 面粉")]
	FOOD, //2 ==== 粮食 -> 面粉
	[InspectorName("SOLDIER - idx[3] - 士兵 -> 面包")]
	SOLDIER, //3 ==== 士兵 -> 面包
}

public class ELevy_TypeComparer : IEqualityComparer<ELevy_Type>{
	public bool Equals(ELevy_Type x, ELevy_Type y) { return x == y; }
	public int GetHashCode(ELevy_Type obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 4;
}
}

File: game_server/ServerProtocol/src/Common/LevyEnum/ELevy_Type.java
package Common.LevyEnum;

/*********
 * 征收类型类型
 **/
public enum ELevy_Type {
	NONE, //0 ==== 
	SILVER, //1 ==== 银币 -> 金币
	FOOD, //2 ==== 粮食 -> 面粉
	SOLDIER, //3 ==== 士兵 -> 面包
	;
public static final ELevy_Type[]  ELevy_Type_Values = ELevy_Type.values();
public static final int ELevy_Type_Length = ELevy_Type_Values.length;
public static ELevy_Type ELevy_Type_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= ELevy_Type_Length){ return null; }
	return ELevy_Type_Values[_ivalue];
}
}

